include/dxc/DXIL/DxilResource.h

```diff
@@ -29,8 +29,19 @@ class DxilResource : public DxilResourceBase {
   static unsigned GetNumDimensionsForCalcLOD(Kind ResourceKind);
   /// Total number of offsets (in [-8,7]) necessary to access resource.
   static unsigned GetNumOffsets(Kind ResourceKind);
-  /// Whether the resource kind is texture.
+  /// Whether the resource kind is a texture. This does not include
+  /// FeedbackTexture.
   static bool IsAnyTexture(Kind ResourceKind);
+  /// Whether the resource kind is an array of textures. This does not include
+  /// texture feedback arrays.
+  static bool IsAnyTextureArray(Kind ResourceKind);
+  /// Whether the resource kind is a texture cube or an array of texture cubes.
+  static bool IsAnyTextureCube(Kind ResourceKind);
+  /// Whether the resource kind is a FeedbackTexture, or an array of
+  /// FeedbackTexture.
+  static bool IsFeedbackTexture(Kind ResourceKind);
+  /// Whether the resource kind is an array of any kind of resource.
```

Comment:
Can we replace "an array of any kind of resource" with "a texture or feedback texture kind with an array dimension"?